scraper/src/github-scraper/types.ts

```diff
@@ -0,0 +1,240 @@
+    email: string;
+  };
+  url: string;
+  message: string;
+}
+
+interface GitHubEvent {
+  id: string;
+  actor: Actor;
+  repo: {
+    id: number;
+    name: string;
+    url: string;
+  };
+  public: boolean;
+  created_at: string;
+  org: Actor;
+}
+
+interface PullRequestReviewEvent extends GitHubEvent {
+  type: "PullRequestReviewEvent";
+  payload: {
+    action: string;
+    review: Review;
+    pull_request: PullRequest;
+  };
+}
+
+interface IssuesEvent extends GitHubEvent {
+  type: "IssuesEvent";
+  payload: {
+    action: string;
+    issue: Issue;
+  };
+}
+
+interface IssueCommentEvent extends GitHubEvent {
+  type: "IssueCommentEvent";
+  payload: {
+    action: string;
+    issue: Issue;
+    comment: Comment;
+  };
+}
+
+export interface PullRequestEvent extends GitHubEvent {
+  type: "PullRequestEvent";
+  payload: {
+    action: string;
+    pull_request: PullRequest;
+  };
+}
+
+export type IGitHubEvent =
+  | PullRequestReviewEvent
+  | IssuesEvent
+  | IssueCommentEvent
+  | PullRequestEvent;
+
+export interface ActivityData {
+  last_updated?: string;
+  activity: Activity[];
+  open_prs: OpenPr[];
+  pr_stale?: number;
+  authored_issue_and_pr: AuthoredIssueAndPr[];
+  discussions?: any;
+}
+export interface ProcessData {
+  [key: string]: ActivityData;
+}
+
+export const ACTIVITY_TYPES = [
+  "comment_created",
+  "issue_assigned",
+  "issue_closed",
+  "pr_reviewed",
+  "issue_opened",
+  "eod_update",
+  "pr_opened",
+  "pr_merged",
+  "pr_collaborated",
+] as const;
+
+export interface Action {
+  event: string;
+  source: {
+    type: string;
+    issue: {
+      pull_request: boolean;
+      repository: {
+        full_name: string;
+      };
+      user: {
+        login: string;
+      };
+      number: number;
+    };
+  };
+  assignee: {
+    login: string;
+  };
+  created_at: Date;
+}
+export interface Activity {
+  type: (typeof ACTIVITY_TYPES)[number] | string;
+  title: string;
+  time: string;
+  link: string;
+  text: string;
+  collaborated_with?: string[];
+  turnaround_time?: number;
+}
+
+export interface OpenPr {
+  link: string;
+  title: string;
+  stale_for: number;
+  labels: string[];
+}
+
+export interface AuthoredIssueAndPr {
+  issue_link: string;
+  pr_link: string;
+}
+
+export type Discussion = {
+  node: {
+    id: string;
+    title: string;
+    author: {
+      login: string;
+      avatarUrl: string;
+    };
+    url: string;
+    category: {
+      id: string;
+      name: string;
+      emoji: string;
+    };
+    upvoteCount: number;
+    reactions: {
+      totalCount: number;
+    };
+    comments: {
+      edges: {
+        node: {
+          author: {
+            login: string;
+            avatarUrl: string;
+          };
+          upvoteCount: number;
+          isAnswer: boolean;
+        };
+      }[];
+    };
+    createdAt: string;
+    isAnswered: boolean;
+  };
+};
+
+export type Edge = {
+  node: {
+    discussions: {
+      edges: Discussion[];
+    };
+  };
+};
```

Comment:
Either give a more meaningful name for the type or  inline the type definition in the place where it is used as it is used only in one place and is not re-used elsewhere.